### ISSUE TITLE
1485 Potentially new circuit/engine/save crash

### ISSUE THREAD

**Issue by GoogleFrog** [beyond-all-reason/spring#595]
https://steamcommunity.com/app/334920/discussions/0/3766732914509042579/

**Comment by GoogleFrog** [comment]
The AI is whatever was built with 1344.

**Comment by lhog** [comment]
Please avoid mixing AI and engine versions. Both projects are built at the same time and stack translation utilities assume versions are in-sync between the components.

**Comment by GoogleFrog** [comment]
Right. Adding new AI versions is a massive pain though because I have to keep compatibility with previous engine versions.

**Comment by GoogleFrog** [comment]
Here are steps to reproduce, but I tried for 16 minutes and didn't get a crash.
 - Play on Brutal on planet Quora Rosia Dyo
 - Apparently it randomly crashes in about five minutes, sometimes it doesn't though.

Here is the planet:
![image](https://user-images.githubusercontent.com/1318249/214729189-042ff005-f60d-47b9-a214-e49d746518a3.png)

You can reach it with Singleplayer = "Zero-K Dev" in the Developer tab of settings and "Debug auto win" enabled.

**Comment by GoogleFrog** [comment]
[_script.txt](https://github.com/beyond-all-reason/spring/files/10505056/_script.txt)
This script should also run the mission.

**Comment by lhog** [comment]
> Right. Adding new AI versions is a massive pain though because I have to keep compatibility with previous engine versions.

Can you switch to relying on Circuit version shipped with the engine (and get rid of a separate one)?

**Comment by GoogleFrog** [comment]
Not unless the engine is going to include six copies of circuit with my difficulty level config files.
